### PR TITLE
Datasource-http: Improve error handling and error messages

### PR DESCRIPTION
**Pull request by xnyo**
Improves error handling and error messages in the datasource-http example.
No 207-related changes to do, because the datasource does not have a backend.

Related to #109

### CODE REVIEW

File: examples/datasource-http/src/DataSource.ts [hunk 1]
```diff
@@ -41,11 +41,20 @@ export class DataSource extends DataSourceApi<MyQuery, MyDataSourceOptions> {
        * }
        */
       const datapoints = response.data.datapoints;
+      if (typeof datapoints === 'undefined') {
```
Comment (academo) [inline examples/datasource-http/src/DataSource.ts:44]: use 

```
if (datapoints === undefined) {
```

instead.

small explanation: `typeof [var] === 'undefined'` is a bit less performant and it is only useful when you are not sure if the variable will be initialized or not. In this case you are sure the variable is initialized (in the previous line) and you can use the small optimization of comparing directly to undefined vs getting the type and comparing strings.

File: examples/datasource-http/src/DataSource.ts [hunk 2]
```diff
@@ -41,11 +41,20 @@ export class DataSource extends DataSourceApi<MyQuery, MyDataSourceOptions> {
...
+        throw new Error('Remote endpoint reponse does not contain "datapoints" property.');
+      }
 
       const timestamps: number[] = [];
       const values: number[] = [];
 
       for (let i = 0; i < datapoints.length; i++) {
+        if (typeof datapoints[i].Time === 'undefined') {
```
Comment (academo) [inline examples/datasource-http/src/DataSource.ts:52]: In this specific case you could argue, that `Time` might not be initialized but you are actually accessing a property of an already initialized variable `datapoints[i]` that might not exist and thus return undefined.

File: examples/datasource-http/src/DataSource.ts [hunk 3]
```diff
@@ -93,7 +102,7 @@ export class DataSource extends DataSourceApi<MyQuery, MyDataSourceOptions> {
       if (_.isString(err)) {
         message = err;
       } else if (isFetchError(err)) {
-        message += err.statusText ? err.statusText : defaultErrorMessage;
+        message = `Fetch error: ${err.statusText ? err.statusText : defaultErrorMessage}`;
```
Comment (academo) [inline examples/datasource-http/src/DataSource.ts:105]: Nit: re-write this to make it more readable, one might think this ternary condition will always return true if you don't look carefully:


```
message = "Fetch error: " + err.statusText ? err.statusText : defaultErrorMessage;
```

**Review by academo — APPROVED** [review]
I added a few nits to make the code neater but good to go if you want to skip them.

**Comment by xnyo** [comment]
Thanks for the review @academo , I have applied your suggestions :smiley:

**Pull request merged**